common/bolt11.c

```diff
@@ -715,7 +715,6 @@ struct bolt11 *bolt11_decode_nosig(const tal_t *ctx, const char *str,
 	memset(have_field, 0, sizeof(have_field));
 	b11->routes = tal_arr(b11, struct route_info *, 0);
 
-	assert(!has_lightning_prefix(str));
```

Comment:
I think this is not right. or better the fix should be some where else.

The basic idea to have the assertion here is specified inside my commit description here 5f6642a6ff6679503d3c0cbbdef4a791335bd6af

In summary 

> To address this issue, we need to strip the `lightning:` prefix before
    calling each core lightning command. In addition, we should
    modify the invstring inside the db with the canonical one.

The fuzz test shows only that my `invstr_param` callback is not exhaustive and IMHO we should change it and not remove the assertion.

The assertion shows only the bug

Reply:
A possible check if do some invalid character check inside the string, in this way the the value will never pass `invstr_param`, and the assert is never trigger

Reply:
No, it's exactly right.

to_canonical_invstr() strips "lightning:", and that is all!  But the remainder could still start with "lightning:", or any other junk.  We shouldn't even check, and let it fail as expected because it's not valid bech32.